[RDY] vim-patch:8.0.0286,8.1.{674,683}

janlazo opened the pull request:
**vim-patch:8.0.0286: not always redrawing after screen resize**

Problem:    When concealing is active and the screen is resized in the GUI it
            is not immediately redrawn.
Solution:   Use update_prepare() and update_finish() from
            update_single_line().
https://github.com/vim/vim/commit/c10f0e7cb0f35eea489b038e56c87b818eee975b

**vim-patch:8.1.0674: leaking memory when updating a single line**

Problem:    Leaking memory when updating a single line.
Solution:   Do not call start_search_hl() twice.
vim/vim@6d5b4f5

**vim-patch:8.1.0683: spell highlighting does not always end**

Problem:    Spell highlighting does not always end. (Gary Johnson)
Solution:   Also reset char_attr when spell errors are highlighted.
vim/vim@637532b

Close https://github.com/neovim/neovim/pull/8853

janlazo committed:
vim-patch:8.0.0286: not always redrawing after screen resize

Problem:    When concealing is active and the screen is resized in the GUI it
            is not immediately redrawn.
Solution:   Use update_prepare() and update_finish() from
            update_single_line().
https://github.com/vim/vim/commit/c10f0e7cb0f35eea489b038e56c87b818eee975b

janlazo commented:
https://gist.github.com/janlazo/cea7acf2711e89a9bc4f7c79612c6e54#regex is likely required. Not sure about 8.0.0588

justinmk commented:
Why are those later patches required? Do they fix issues that were introduced by 8.0.0286?

ckelsel commented:
#7588

janlazo commented:
@ckelsel I know. I opened this PR to include related patches later.

janlazo commented:
This patch should work after https://github.com/neovim/neovim/pull/8662 and 8.0.0647.

janlazo commented:
Actually, this patch is likely coupled with patches from #8662 so I'll close this now.

janlazo committed:
vim-patch:8.1.0674: leaking memory when updating a single line

Problem:    Leaking memory when updating a single line.
Solution:   Do not call start_search_hl() twice.
https://github.com/vim/vim/commit/6d5b4f566a2a50c1de7300336e9e4f5e761500a8

janlazo committed:
vim-patch:8.1.0683: spell highlighting does not always end

Problem:    Spell highlighting does not always end. (Gary Johnson)
Solution:   Also reset char_attr when spell errors are highlighted.
https://github.com/vim/vim/commit/637532b3c0ca41f0de7e90b6f3c0defe06369372

justinmk commented:
@janlazo Thank you very much for solving this mystery!
